### DISCUSSION TITLE
Status of Jamulus integration in Linux distributions

### DISCUSSION THREAD

**Discussion by corrados** [jamulussoftware/jamulus#914] [Announcements]
This issue is basically just a list of URLs which relate to possible integrations of Jamulus in Linux distributions.

Several years ago, there was an attempt to include Jamulus in Redhat:
https://bugzilla.redhat.com/show_bug.cgi?id=1118963
The CONFIG options "noupcasename" and "opus_shared_lib" were a result of it.

More recently, mirabilos did some work for including Jamulus in Debian:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=958146
We also have Jamulus Issues related to that, e.g., https://github.com/corrados/jamulus/issues/100

It seems that there is a Jamulus package in OpenSuse available:
https://software.opensuse.org/package/Jamulus

If you know of other Linux distributions or have updated informations, please post them here.

<a href="https://repology.org/project/jamulus/versions">
    <img src="https://repology.org/badge/vertical-allrepos/jamulus.svg" alt="Packaging status">
</a>

**Comment by trebmuh** [comment]
Not a mainstream distribution, but one focused on music tools, jamulus is actually in its testing phase for [LibraZiK](https://librazik.tuxfamily.org/doc3/). I'm waiting to finish the whole French translation to get it out of the _testers_ repo and in the _users_ repo.

**Comment by Snayler** [comment]
ArchLinux (and any Arch based distro) has the latest version available in the AUR.
https://aur.archlinux.org/packages/jamulus/

**Reply by jujudusud** [reply]
There is also a 'headless' package : 
https://aur.archlinux.org/packages/jamulus-headless/

**Comment by pljones** [comment]
Just noticed the AUR has its own `jamulus.desktop`.  Maybe the Linux distros could agree on a `jamulus.desktop` entry that can be included or have one generated that's distro-specific by the build process?

**Comment by trebmuh** [comment]
Yep, distros tend to do that from time to time. Sometimes with a very good reason, but usually not. Maybe someone can drop them a message about the [existence of a upstream desktop file](https://github.com/corrados/jamulus/blob/master/distributions/jamulus.desktop) ?

Oh, after checking, I know the packager and will report that to him.

**Comment by jujudusud** [comment]
Hi all, Co-maintainer of AUR Archlinux package here.
just finishing to update the package in order to use upstream jamulus.desktop file.
Thanks a lot trebmuh and all of you guy for all that stuff ! :-)

**Comment by jtojnar** [comment]
It would be nice if the desktop file was installed by qmake, see https://github.com/corrados/jamulus/pull/247#discussion_r428560108, https://github.com/corrados/jamulus/pull/247#issuecomment-633315300 and https://github.com/corrados/jamulus/pull/247#issuecomment-633318364.



P.S. For cross-distro tracking there is https://repology.org/project/jamulus/versions.

**Comment by corrados** [comment]
Can you create a pull request for that? If not you, than anybody else?

**Comment by corrados** [comment]
BTW: The link in your post is pretty cool :-)

_Edit: I have moved the link to the packaging status in the first post of this Issue._

**Comment by jtojnar** [comment]
Unfortuntaely, I will not be able to open a PR anytime soon.

**Comment by corrados** [comment]
@jtojnar atsampson was so kind to do the implementation and the new code is already merged to the Git master. Do you have the possibility to check if the implementation works as you requested?

**Comment by seb314** [comment]
For NixOs/Nix there's a [package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/jamulus/default.nix) on master (but not in any of the stable releases currently)

**Comment by seb314** [comment]
> @jtojnar [...] Do you have the possibility to check if the implementation works as you requested?

https://github.com/corrados/jamulus/pull/331#issuecomment-639666652

**Comment by trebmuh** [comment]
As said [in #439](https://github.com/corrados/jamulus/pull/349#issue-432650522), jamulus 3.5.6 is now officially in the LibraZiK-3 users repo.

**Comment by ann0see** [comment]
I just found jamulus in homebrew for macOS: https://formulae.brew.sh/cask/jamulus

**Comment by cosmicfranz** [comment]
Hi everybody, I’m new to Jamulus and I would like to try it on a private Fedora Linux server (version 32). I have been able to create a RPM package, loosely based on the OpenSUSE spec file. I have installed such package and I can run the executable both as a client and a server.

So far, so good. However, a problem arises if I try to connect to this server with a client running on the same machine: in other words, I have two instances of the same executable, one started with ‘-s’. When I ask the client to connect to localhost, it crashes immediately with this error:

_Fatal (internal) error in celt/celt_decoder.c, line 126: assertion failed: st->end <= 21_

which seems related to the OPUS library I am using, that is, the one already present on my Fedora installation (version 1.3.1). I have chosen to use the system library instead of the bundled one at compile time.

I know nothing about opus library, and very little about Jamulus internals. Any pointer on how to solve this issue?

Thanks
Francesco Napoleoni


The following is the spec file I have written, on its way to conform to Fedora guidelines:


%define _debugsource_template %{nil} # this to get around a build error

Summary:    Low-latency internet connection tool for real-time jam sessions
Name:       jamulus
Version:    3.6.0
Release:    1
URL:        https://jamulus.io/
License:    GPLv2 and STK
Source:     %{name}-%{version}.tar.gz
Patch0:     get_rid_of_bundled_opus_library.patch # Deletes the libs/opus directory
Patch1:     tweak_license_file.patch # Removes the OPUS license and puts the STK license on another file. But this is still not enough for rpmlint...


BuildRequires: qt5-qtdeclarative-devel
BuildRequires: jack-audio-connection-kit-dbus
BuildRequires: jack-audio-connection-kit-devel
BuildRequires: opus-devel
BuildRequires: desktop-file-utils


%description
The Jamulus software enables musicians to perform real-time jam sessions over
the internet. There is one server running the Jamulus server software which
collects the audio data from each Jamulus client software, mixes the audio data
and sends the mix back to each client.


%prep
%setup -q -n Jamulus3.6.0
%patch0 -p1
%patch1 -p1


%build
qmake-qt5 PREFIX=/usr CONFIG+=noupcasename CONFIG+=opus_shared_lib
make


%install
install -D -m 0755 %{name} %{buildroot}/%{_bindir}/%{name}
desktop-file-install --dir=%{buildroot}/%{_datadir}/applications %{_builddir}/Jamulus3.6.0/distributions/%{name}.desktop
install -D -m 0644 distributions/%{name}.png %{buildroot}/%{_datadir}/icons/hicolor/512x512/%{name}.png


%files
%defattr(0644, root, root, 0755)
%attr(0755, -, -) %{_bindir}/%{name}
%{_datadir}/applications/%{name}.desktop
%{_datadir}/icons/hicolor/512x512/%{name}.png
%doc README.md CONTRIBUTING.md ChangeLog
%license COPYING COPYING-stk

**Comment by atsampson** [comment]
Yes, it's because you're using the system libopus - there's a bug in libopus's custom mode validation [that's been fixed in Git](https://gitlab.xiph.org/xiph/opus/-/commit/06a651547f4a197c29b6abc23b530833855ee358) but hasn't made it into a release yet.

**Comment by cosmicfranz** [comment]
Thank you. I have just verified that compiling with the internal opus library actually solves this problem.

Aside from this, after some research I have found an unofficial RPM for Fedora made by Yann Collette, which can be found [here](https://copr-be.cloud.fedoraproject.org/results/ycollet/linuxmao/fedora-32-x86_64/01415650-Jamulus/), version 3.5.5.

cheers
Francesco Napoleoni

**Comment by yajo** [comment]
It seems @ycollet volunteered to [package it for Fedora](https://bugzilla.redhat.com/show_bug.cgi?id=1844120). I feel happier now, kudos! :heart: 

Oops I just noticed I'm repeating the same as https://github.com/corrados/jamulus/issues/223#issuecomment-722304114... shame on me :see_no_evil:

**Comment by trebmuh** [comment]
Jamulus 3.6.1 is now officially in the LibraZiK-3 users repo.

**Comment by trebmuh** [comment]
Jamulus 3.6.2 has been in the LibaZiK-3 users repo since mid-December.

**Comment by ann0see** [comment]
I converted this to a discussion now since this is not an issue.

**Comment by pljones** [comment]
Does https://github.com/kdoren/jambox-pi-gen Jambox count as a distribution?  Pre-built images for running Jamulus and SonoBus are provided.

**Reply by kdoren** [reply]
I'm biased, but I would call Jambox for Raspberry Pi a distribution.  I do my own builds of not only Jamulus but also jack audio,  alsa-utils, and kernels which are served from the jambox-project apt repo.  New Jamulus releases show up there almost immediately, and can be easily pulled using apt (there's even desktop icon to update apps).

Currently using Jamulus 3.7.0
https://github.com/kdoren/jambox-pi-gen

Repo instructions are here for anyone who just wants the Jamulus package (but I strongly recommend using the full Jambox image, which has little-to-no assembly required and is tuned as a system to run Jamulus well on a Raspberry Pi).  https://github.com/kdoren/jambox-debs

**Comment by mirabilos** [comment]
Jamulus is now up-to-date in [Debian](https://packages.debian.org/search?keywords=jamulus) unstable (will migrate to testing in a couple of days and then become a part of the next stable 11 “bullseye” at which time I can backport it) and Ubuntu hirsute (with backports to previous LTS releases available from [my PPA](https://launchpad.net/~mirabilos/+archive/ubuntu/ppa)).

**Reply by mirabilos** [reply]
I’ve submitted Jamulus to buster-backports. It will be available after backports-ftpmaster review, which is mostly dependent on their time; I expect no issues.

The PPA version number is one smaller than the one in Debian, but that has organisatorical reasons and is of no import. The next upload that actually changes something will get them into sync again.

**Reply by mirabilos** [reply]
And voilà, https://packages.debian.org/search?keywords=jamulus also shows the buster-backports one, so it’s installable for end users.

**Comment by NexAdn** [comment]
I've written a Gentoo Ebuild for Jamulus. It's currently located in my own overlay, but maybe it can be integrated in some larger overlay later on:
https://gitlab.com/NexAdn/nex-overlay/-/tree/master/media-sound/jamulus

**Reply by NexAdn** [reply]
version 3.7.0 is now available in my overlay

**Reply by NexAdn** [reply]
I've moved my ebuild to the fem-overlay https://gitlab.fem-net.de/gentoo/fem-overlay

This overlay is also listed on the Gentoo website and thus can be added using eselect-repository

**Comment by astralchan** [comment]
Release 3.7.0 is now on VOID :3

**Comment by jujudusud** [comment]
Jamulus is available for Archlinux users at 3.7.0 version.
Thanks a lot for all that Amazing Stuff !

**Comment by ycollet** [comment]
Jamulus 3.7.0 is available for Fedora 33 and 34 (64 bits only) via this COPR repository:
https://copr.fedorainfracloud.org/coprs/ycollet/linuxmao/

**Comment by trebmuh** [comment]
3.7.0 is in LibraZiK-3 as of now

**Comment by yajo** [comment]
FWIW, it's available on https://flathub.org/apps/details/io.jamulus.Jamulus, for all distros at once. Notice that you must be using `pipewire-jack-audio-connection-kit` on the host (so that might reduce temporarily the supported distros in practice).

**Comment by trebmuh** [comment]
3.8.0 is in LibraZiK-3 as of now

**Comment by jujudusud** [comment]
3.8.2 Jamulus app + headless are up to date for Archlinux in AUR.

**Reply by jujudusud** [reply]
Hi everyone.
Announcement: There is a new Pro Audio repository for archlinux where Jamulus x86_64 is present: https://arch.osamc.de/#x86_64/jamulus
Note there is also a aarch64 package.

If you are interested, you can have a look here: https://wiki.archlinux.org/title/Unofficial_user_repositories#proaudio

**Reply by ann0see** [reply]
Great. If you want, you can edit the KB page on the website for alternative installation methods

**Reply by jujudusud** [reply]
Ok thanks ann0see. You will be happy to read this in the # 801 PR on Jamuluswebsite repository. :-)

**Comment by mirabilos** [comment]
3.8.2 entered Debian: https://packages.qa.debian.org/j/jamulus/news/20220225T191926Z.html

**Reply by ann0see** [reply]
That’s great! 
*And in my opinion a reason to mention on the client install page* We moved away from it unfortunately…

**Reply by ann0see** [reply]
@mirabilos could you please open an issue and explain the reason of https://evolvis.org/plugins/scmgit/cgi-bin/gitweb.cgi?p=alioth/jamulus.git;a=blob;f=debian/patches/fix-help-stderr.diff;h=7ebf8692b94fc2d0bcd0805b42b5d9ba7af58c86;hb=e9f32e091458d36da26c0ae39b9c9498a7258ba1

**Reply by ann0see** [reply]
Ok. Saw the link where you mentioned it. I think you should still raise an issue.

**Comment by jujudusud** [comment]
In archlinux, 3.9.0 is now available for regular and headless version directly in binary from the [arch osam repository](https://arch.osamc.de/#x86_64/jamulus).
It is available for x86_64 and aarch64 too.
It uses perfectly well the Qt version 6.

**Comment by jujudusud** [comment]
In archlinux, 3.9.1 is now available for regular and headless version directly in binary from the [arch osam repository](https://arch.osamc.de/#x86_64/jamulus).
It is also available in AUR repository.
It is available for x86_64 and aarch64 too.
Headless users should install qt6-base-headless package from AUR instead of the regular qt6-base one in order to reduce installed stuff on a server.

**Comment by trebmuh** [comment]
Hi all
3.10.0 is in LibraZiK-4 as of now
